com.avaloq.tools.ddk.xtext.builder/src/com/avaloq/tools/ddk/xtext/builder/MonitoredClusteringBuilderState.java

```diff
@@ -708,6 +708,12 @@ protected void storeBinaryResource(final Resource resource, final BuildData buil
           }
         }, binaryStorageExecutor);
       } else {
+        CompletableFuture.runAsync(() -> {

```

Comment:
@gerdleon : There should not be because we do not remove the uri from the source level uri list (as we do in the if branch).
Maybe you wonder now how the bug can be triggered then, and I think it could be triggered if the binary model is accessed during any build following the one where we failing the save action